### PR TITLE
rwx: add "Configuring Volume Locality for an RWX Volume"

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#7872

#### Which issue(s) this PR fixes:

Issue #

### CODE REVIEW

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 1]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
 2. For manually created Longhorn volumes (restore, DR volume) the access mode can be specified during creation in the Longhorn UI.
 3. When creating a PV/PVC for a Longhorn volume via the UI, the access mode of the PV/PVC will be based on the volume's access mode.
 4. One can change the Longhorn volume's access mode via the UI as long as the volume is not bound to a PVC.
-5. For a Longhorn volume that gets used by a RWX PVC, the volume access mode will be changed to RWX.
+5. For a Longhorn volume that gets used by an RWX PVC, the volume access mode will be changed to RWX.
 
+## Configuring Volume Locality for an RWX Volume
+
+  Longhorn introduces several enhancements to control the volume locality of RWX volumes. This is accomplished by determining the location of the share manager pod associated with an RWX volume. This section explains how to configure the data locality feature for RWX volumes.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:47]:
```suggestion
## Configuring Volume Locality for RWX Volumes

Longhorn provides new settings that allow you to precisely control the data locality of RWX volumes (through identification of associated Share Manager pods). These granular settings work with related global settings to provide optimal performance, resilience, and adherence to organizational policies or constraints.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 2]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+
+### Using `shareManagerNodeSelector`
+
+You can specify the `shareManagerNodeSelector` in the `storageClass.Parameters` to dictate on which nodes RWX volumes can be scheduled. These selectors are merged with global `system-managed-components-node-selector` settings and applied to the share manager pods of the RWX volumes. This provides more control over the locality of the RWX volumes.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:51]:
```suggestion
### `shareManagerNodeSelector`

You can use the StorageClass parameter `shareManagerNodeSelector` to specify selectors for identifying nodes that RWX volumes can be scheduled on. These selectors are merged with global `system-managed-components-node-selector` settings and then applied to the Share Manager pods of the RWX volumes to provide more control over volume locality.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 3]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+  ```
+  kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  parameters:
+    shareManagerNodeSelector: label-key1:label-value1;label-key2:label-value2
+  ```
+  In the above example, RWX volumes provisioned with this StorageClass will be scheduled on nodes that have the label key=val.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:63]:
```suggestion
  In this example, RWX volumes provisioned with the specified StorageClass will be scheduled on nodes with the label `key=val`. 
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 4]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  parameters:
+    shareManagerNodeSelector: label-key1:label-value1;label-key2:label-value2
+  ```
+  In the above example, RWX volumes provisioned with this StorageClass will be scheduled on nodes that have the label key=val. 
+
+### Using `allowedTopologies`
+
+Longhorn converts the `storageClass.allowedTopologies` settings into affinity rules for the share manager pods of the RWX volumes. This ensures the pods are scheduled on nodes that meet the specified topological requirements, such as regions or zones, aligning with the RWX volume locality.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:67]:
```suggestion
### `allowedTopologies`

Longhorn converts the `storageClass.allowedTopologies` settings into affinity rules for the Share Manager pods of the RWX volumes. This ensures that the pods are scheduled on nodes that meet the specified topological requirements (such as regions and zones) and align with the RWX volume locality.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 5]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+  metadata:
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  allowedTopologies:
+  - matchLabelExpressions:
+    - key: topology.kubernetes.io/region
+      values:
+      - us-west-1
+  ```
+  In the above example, the share manager pods and RWX volumes will be scheduled in the `us-west-1` region.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:82]:
```suggestion
  In this example, the Share Manager pods and RWX volumes will be scheduled in the `us-west-1` region.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 6]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+
+### Using `shareManagerTolerations`
+
+You can define `shareManagerTolerations` in the `storageClass.Parameters`. These tolerations are merged with global `taint-toleration` settings and applied to the share manager pods, allowing more flexible scheduling based on node taints.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:86]:
```suggestion
### `shareManagerTolerations`

You can also use the StorageClass parameter `shareManagerTolerations` to allow more flexible scheduling based on node taints. The defined tolerations are merged with global `taint-toleration` settings and then applied to the Share Manager pods.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 7]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+  ```
+  kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  parameters:
+    shareManagerTolerations: nodetype=storage:NoSchedule
+  ```
+  In the above example, the share manager pods will tolerate the `nodetype=storage:NoSchedule` taint on nodes, allowing them to be scheduled on those nodes.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:98]:
```suggestion
  In this example, the Share Manager pods will tolerate the `nodetype=storage:NoSchedule` taint on nodes, allowing them to be scheduled on those nodes.
```

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md [hunk 8]
```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
...
+
+  By combining these three configurations, you can precisely control the data locality of RWX volumes based on your requirements, ensuring optimal performance, resilience, and adherence to organizational policies or constraints.
```
Comment (jillian-maroket) [inline content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md:100]: Please remove this. I moved the information to the introductory paragraph.

**Review by jillian-maroket — COMMENTED** [review]
@derekbit Let me know if you have concerns about the changes.

File: content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md (hunk 8 above)
Comment (derekbit) [reply content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md]: Removed. Thank you.

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Review by jillian-maroket — APPROVED** [review]

**Pull request merged**